### PR TITLE
LPS-158146 Don't allow drop inside form without mapping

### PR DESCRIPTION
**Pull request by sandrodw3**
cc @victorg1991

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/drag-and-drop/defaultComputeHover.js [hunk 1]
```diff
@@ -300,6 +298,19 @@ function getItemPosition(item, monitor, targetRefs, orientation) {
 	];
 }
 
+function shouldBeIgnoredInElevation(item) {
+
+	// Dropping inside a collection or inside a row is illegal
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/drag-and-drop/defaultComputeHover.js:308]: Some typos here, you reminded me of someone :trollface:

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/drag-and-drop/useDragAndDrop.js [hunk 2]
```diff
@@ -340,7 +341,26 @@ export function DragAndDropContextProvider({children}) {
 }
 
 function computeDrop({dispatch, layoutDataRef, onDragEnd, state}) {
-	if (state.droppable && state.dropItem && state.dropTargetItem) {
+	if (!state.droppable) {
+		if (state.dropItem.fragmentEntryType === FRAGMENT_ENTRY_TYPES.input) {
+			openToast({
+				message: Liferay.Language.get(
+					'Error: Form components can only be placed inside a mapped Form Container.'
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/drag-and-drop/useDragAndDrop.js:348]: This is my new favorite language key ❤️

**Review by sandrodw3 — COMMENTED** [review]
@victorg1991 this is super cool! 🥰 you look like a good candidate to be the new Lord of DnD. Just left a comment about a language key. I also found two cases that are not working well, at least for me:

- Dropping an input fragment inside an empty column is not showing red feedback when dragging
- The red feedback remains after drop in some cases, looks like if the state was not being cleared after it

By the way, I rebased this over https://github.com/liferay-echo/liferay-portal/pull/8982, as it also caused a conflict (I resolved it). I hope my changes didn't cause any of the problems I mentioned 🙏

**Comment by victorg1991** [comment]
resent here: https://github.com/liferay-echo/liferay-portal/pull/9000

**Pull request closed without merging**